packages/react-wildcat-prefetch/src/index.js

```diff
@@ -65,39 +65,69 @@ function getDisplayName(Comp) {
+function prefetchWrap(action, options = {}) {
+    const key =
+        options.key ||
+        (typeof options === "string" ? options : __DEFAULT_ASYNC_DATA_KEY__);
+    let initialDataKey = options.initialDataKey || __DEFAULT_INITIAL_DATA_KEY__;
 
     return function prefetchWrapper(ComposedComponent) {
         action = action || ComposedComponent[__DEFAULT_STATIC_METHOD__];
-        var _action = getAction(action, ComposedComponent);
+        const _action = getAction(action, ComposedComponent);
+
+        class Prefetch extends React.Component {
+            state = {};
+
+            static propTypes = {
+                [initialDataKey]: React.PropTypes.string,
+                location: React.PropTypes.shape({
+                    pathname: React.PropTypes.string.isRequired
+                }).isRequired
+            };
+
+            static prefetch = {
+                run: function run(props) {
+                    return _action(props);
+                },
 
-        var Prefetch = React.createClass({
-            propTypes: {
-                [initialDataKey]: React.PropTypes.string
-            },
+                getKey: function getKey() {
+                    return key;
+                },
 
-            componentWillMount: function componentWillMount() {
-                var canUseDOM = (typeof options.canUseDOM !== "undefined") ? options.canUseDOM : ExecutionEnvironment.canUseDOM;
+                setInitialDataKey: function setInitialDataKey(
+                    customInitialDataKey
+                ) {
+                    initialDataKey = customInitialDataKey;
+                }
+            };
+
+            static WrappedComponent = ComposedComponent;
+            static displayName = `Prefetch(${getDisplayName(ComposedComponent)})`;
+
+            componentWillMount() {
+                const canUseDOM = typeof options.canUseDOM !== "undefined"
+                    ? options.canUseDOM
+                    : ExecutionEnvironment.canUseDOM;
 
                 /* istanbul ignore else */
                 if (canUseDOM) {
-                    var initialDataID = this.props[initialDataKey] || initialDataKey;
+                    const initialDataID =
+                        this.props[initialDataKey] || initialDataKey;
 
-                    var initialData = window[initialDataID] ? {
-                        ...window[initialDataID]
-                    } : undefined;
+                    const initialData = window[initialDataID]
+                        ? {
+                            ...window[initialDataID]
+                        }
+                        : undefined;
 
-                    var newState = {};
+                    const newState = {};
 
                     if (initialData) {
                         // Delete stored objects
-                        if (window[initialDataID] && window[initialDataID][key]) {
+                        if (
+                            window[initialDataID] &&
```

Comment:
Is this executed on the server side at all? If so we should check that `window` is an object with `window &&`